### DISCUSSION TITLE
[ALL] 회원 가입 시, 응답으로 location을 무엇을 넘겨줄까?

### DISCUSSION THREAD

**Discussion by kth990303** [woowacourse-teams/2022-nae-pyeon#81] [ALL 설계 이슈]
현재 회원 가입 시, 응답으로 아래와 같이 넘겨주고 있습니다.
```javascript
201 CREATED

header: {
	location: /api/v1/members/me
} 
```

하지만, 회원 가입 시에 토큰을 넘겨주지도 않고,회원 가입 시에 로그인 페이지로 이동시키는 것이 좋지 않을까 하여 **아래와 같이 변경하면 좋을 것 같은데 어떤가요?**
```javascript
201 CREATED

header: {
	location: /api/v1/login
} 
```

- 추가로, 201 created할 때에는 무조건 id 값을 넘겨주기로 했었습니다. 그렇지만, 사실 상 **회원가입에서는 id를 넘겨주는 것이 필요 없을 것 같은데 어떻게 생각하시나요**?

**Comment by yxxnghwan** [comment]
현재까지 구조에서 로케이션 헤더를 사용하지 않는 것으로 알고 있어요.
넘겨주는 이유는 관례상 생성된 리소스에 접근 가능한 URI를 넘겨주는 것 때문에 넘겨주고 있었는데,
저희가 저번 회의에서 변경하기로 한 구조(/api/v1/members/me)에서는 
생성된 멤버 리소스에 접근하는 URI가 당장은 존재하지 않습니다.(추후에 멤버에 대한 단건 조회가 추가될 수는 있음.)
추후에 멤버 단건 조회 API가 필요해져서 만든다면 그 값을 로케이션에 담아 반환하는 게 맞다고 생각되고, 그 전에는 다른 URI를 반환해도 상관 없을 것 같아요.
추가로 id를 넘겨주는 것은 롤링페이퍼 생성시에 id를 반환해서 바로 그 롤링페이퍼 페이지를 여는 곳에서 사용하고 있고, 생성시에 id를 반환하는 통일성을 지키기 위해 넘겨주고 있었던 것 같은데, 이 부분이 꼭 통일되지 않아도 된다고 생각하면 id반환도 빼도 괜찮을 것 같아요

**Comment by Seungpang** [comment]
현재까지는 딱히 hearder에 location을 사용하고 있지 않고 그렇게 중요한 부분이 아니라고 생각하는데
만약에 회원가입 후 자동로그인을 한다고 하면
```
header: {
	location: /api/v1/login
} 
```
이 구조가 의미가 있을까요? 이거는 프론트쪽에서 어떤 방식으로 그 이후에 풀어 나가냐에 따라 많이 다를 것 같습니다.
